src/MMCPClient.cpp

```diff
@@ -0,0 +1,769 @@
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+#include "MMCPClient.h"
+#include "Host.h"
+#include "MMCPServer.h"
+#include "mudlet.h"
+
+#include "pre_guard.h"
+#include <QHostAddress>
+#include <QTcpSocket>
+#include <QRegularExpression>
+#include <QtGlobal>
+#include "post_guard.h"
+
+QString convertToIPv4(QHostAddress addr)
+{
+    // Check if the address is an IPv4-mapped IPv6 address
+    if (addr.protocol() == QAbstractSocket::IPv6Protocol && addr.isInSubnet(QHostAddress::parseSubnet("::ffff:0:0/96"))) {
+        // Convert to IPv4
+        QHostAddress ipv4addr = QHostAddress(addr.toIPv4Address());
+        return ipv4addr.toString();
+    }
+    // Return the original address if it's not an IPv4-mapped IPv6 address
+    // or conversion is not applicable
+    return addr.toString();
+}
+
+MMCPClient::MMCPClient(Host* host, MMCPServer* serv) : m_state(Disconnected), tcpSocket(this), mpHost(host), server(serv)
+{
+    //auto ignore or ignorelist match?
+    m_isIgnored = (1 == 0 ? true : false);
+
+    //auto private or privatelist match?
+    m_isPrivate = (1 == 0 ? true : false);
+
+    //auto serve or servelist match?
+    m_isServed = (1 == 0 ? true : false);
+
+    m_isServing = false;
+    m_canSnoop = false;
+    m_isSnooped = false;
+    m_isSnooping = false;
+    m_group = tr("<none>");
+    m_chatName = "";
+
+    //Disable Nagle's algorithm
+    tcpSocket.setSocketOption(QAbstractSocket::LowDelayOption, 1);
+
+    connect(&tcpSocket, &QTcpSocket::connected, this, &MMCPClient::slotConnected);
+    connect(&tcpSocket, &QTcpSocket::disconnected, this, &MMCPClient::slotDisconnected);
+    connect(&tcpSocket, &QTcpSocket::readyRead, this, &MMCPClient::slotReadData);
+#if QT_VERSION >= 0x051500
```

Comment:
The way that the `QT_VERSION` is encoded means that doesn't mean what you think it does... {IIRC you would probably need `0x050F00` as it is hex encoded} but there is an easier way to code it:
```suggestion
#if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
```

Reply:
Got it

Reply:
Can we "Commit suggestion" this - as the current code is lighting up a build warning (so not fatal) for me (on Qt **5.15.8**)...

Reply:
I've got this fix in my local repo I'll push it as soon as I wrap up a couple more things today